src/Unleash/Utilities/ToggleBootstrapFileProvider.cs

```diff
@@ -0,0 +1,33 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using Unleash.Internal;
+
+namespace Unleash.Utilities
+{
+    public class ToggleBootstrapFileProvider : IToggleBootstrapProvider
+    {
+        private readonly string filePath;
+        private readonly IFileSystem fileSystem;
+
+        public ToggleBootstrapFileProvider(string filePath)
+        {
+            this.filePath = filePath;
+            fileSystem = new FileSystem(Encoding.UTF8);
+        }
+
+        internal ToggleBootstrapFileProvider(string filePath, IFileSystem fileSystem)
+        {
+            this.filePath = filePath;
+            this.fileSystem = fileSystem;
+        }
+
+        public string Read()
+        {
+            if (!fileSystem.FileExists(filePath))
```

Comment:
This might be a personal thing (I may have spent too much time in Python) but I would probably remove this guard here for two reasons:

1) Someone has to handle the potential IOException anyway since the last line of this function reads a file and that can potentially fail for reasons outside of the program's control
2) The incredibly rare race condition where a file is wiped between checking for its existence and obtaining a filehandle is taken care of by the previous case anyway

Reply:
Yeah i think you're right. If you code explicitly for a bootstrapfile to be used you'll probably want an operational exception to let you know when it doesn't exist.